sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,595 @@
+
+    @Override
+    public TableType getJdbcTableType()
+    {
+      return TableType.SYSTEM_TABLE;
+    }
+
+    @Override
+    public Enumerable<Object[]> scan(DataContext root)
+    {
+      final List<Object[]> rows = new ArrayList<>();
+      final Map<String, QueryableDruidServer> serverViewClients = serverView.getClients();
+      for (QueryableDruidServer queryableDruidServer : serverViewClients.values()) {
+        final DruidServer druidServer = queryableDruidServer.getServer();
+        final Map<String, DataSegment> segmentMap = druidServer.getSegments();
+        for (DataSegment segment : segmentMap.values()) {
+          Object[] row = new Object[SEGMENT_SERVERS_TABLE_SIZE];
+          row[0] = druidServer.getHost();
+          row[1] = segment.getIdentifier();
+          rows.add(row);
+        }
+      }
+      return Linq4j.asEnumerable(rows);
+    }
+  }
+
+  static class TasksTable extends AbstractTable implements ScannableTable
+  {
+    private final DruidLeaderClient druidLeaderClient;
+    private final ObjectMapper jsonMapper;
+    private final BytesAccumulatingResponseHandler responseHandler;
+
+    public TasksTable(
+        DruidLeaderClient druidLeaderClient,
+        ObjectMapper jsonMapper,
+        BytesAccumulatingResponseHandler responseHandler
+    )
+    {
+      this.druidLeaderClient = druidLeaderClient;
+      this.jsonMapper = jsonMapper;
+      this.responseHandler = responseHandler;
+    }
+
+    @Override
+    public RelDataType getRowType(RelDataTypeFactory typeFactory)
+    {
+      return TASKS_SIGNATURE.getRelDataType(typeFactory);
+    }
+
+    @Override
+    public TableType getJdbcTableType()
+    {
+      return TableType.SYSTEM_TABLE;
+    }
+
+    @Override
+    public Enumerable<Object[]> scan(DataContext root)
+    {
+      class TasksEnumerable extends DefaultEnumerable<Object[]>
+      {
+        private final JsonParserIterator<TaskStatusPlus> it;
+
+        public TasksEnumerable(JsonParserIterator<TaskStatusPlus> tasks)
+        {
+          this.it = tasks;
+        }
+
+        @Override
+        public Iterator<Object[]> iterator()
+        {
+          throw new UnsupportedOperationException("Do not use iterator(), it cannot be closed.");
+        }
+
+        @Override
+        public Enumerator<Object[]> enumerator()
+        {
+          return new Enumerator<Object[]>()
+          {
+            @Override
+            public Object[] current()
+            {
+              TaskStatusPlus task = it.next();
+              return new Object[]{task.getId(), task.getType(),
+                                  task.getDataSource(),
+                                  task.getCreatedTime(),
+                                  task.getQueueInsertionTime(),
+                                  task.getState(),
+                                  task.getRunnerTaskState(),
+                                  task.getDuration(),
+                                  task.getLocation() != null
```

Comment:
`location` can't be null in `TaskStatusPlus`.

Reply:
removed null check